File: zoo_project_cgi_php/CGI C program/Animals/animals.cpp
/*
*	FILE			: animals.cpp
*	PROJECT			: A - 03 : Animal - Serve – website developed in CGI and PHP(PROG2001)
*	FIRST VERSION	: 2020 - 10 - 24 (Rev.04)
*	AUTHORS			: Dusan Sasic & Kevin Downer
*	DESCRIPTION		: The CGI coded in C for the Animal Zoo project.
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "sys/types.h"

#pragma warning(disable : 4996)		// need to do this to be able to use getenv()


//Properties
void DisplayOutput(char* image, char* path);




int main(void)
{


	char* data;				//The variably that will hold query string enviroment variable
	char name[25];			// C style string to store name
	char animal[15];		// C style string to store animal name


	//Constructing HTML head, title and link to an external css file
	printf("Content-type:text/html\n\n");
	printf("<html><head><title>Animal</title><link rel=\"stylesheet\" href=\"../../../ext-css-style-zoo.css\"> </head>\n");
	


	// Get the environment variable containing passed parameters through the URL.
	data = getenv("QUERY_STRING");
	
	//Finding '&' charachrter and replacing it with a whitspace
	char *found = strchr(data, '&');
	*found = ' ';	



	//Scans the the query string and parses two strings, if both strings passed go inside
	if (sscanf(data, "in_name=%s combo_animals=%s", &name, &animal) == 2)
	{

		//Constructing the HTML body and display the greeting message
		printf("<body\n><div class=\"helloName\" align=\"center\">");
		printf("<br><h1> Hello %s, Look If You Dare!</h1></div>", &name);
		

		//Creating a string representing the relative path to the image file for the animal.
		char path_image[500] = "../../../theZoo/";
		strcat(path_image, animal);
		strcat(path_image, ".jpg");

		//Creating a string representing the relative path to the text file for the animal.
		char path_description[500] = "theZoo/";
		strcat(path_description, animal);
		strcat(path_description, ".txt");


		//Function for displaying text and the image based on the file paths
		DisplayOutput(path_image, path_description);


	}
	else
	{
		//In case the the url string was not read properly 
		printf("<p>Invalid Data!</p>");		//Print a message
	}
	
	

	//Close the HTML tags
	printf("</body></html>");
	//Exit the program
	return 0;
}







/*
* FUNCTION		: DisplayOutput
* DESCRIPTION	: This function opens an image based on the file path specified as the parameter and 
				opens a file for reading through the first parameter path and reads three description
				lines from it. Lines are later displayed in formated HTML page.
* RETURN		: shouldSubmit : bool :  Decides if the required User inputs are present and valid.
* Form sent if all the validation requirements are met.
*/
void DisplayOutput(char *image, char *path)
{


	printf("<div><table align=\"center\">");
	printf("<tr><td align=\"right\" width=\"51 % \"> <div class=\"pictureAnimal\"width: 50%; margin: 0 auto;><img src=\"%s\"></img></div></td>", image);
	

	//Open a file		//someText.txt
	FILE* f;
	char line_title[50];
	char line_overview[350];
	char line_characteristics[350];
	char line_special_traits[350];

	//Opens the fil78
	f = fopen(path, "r");
	if (!f)	//If file not opened
	{
		printf("<br><h1>ERROR!!!!</h1>");
		return;
	}

	//fscanf all the way until the fist \n (modified .txt file lil bit, had to)
	fscanf(f, "%[^\n] %[^\n] %[^\n] %[^\n] ", line_title, line_overview, line_characteristics, line_special_traits);

	fclose(f); //Important to close the file


	
	printf("<td align=\"left\" width=\"51 % \"><div id=\"detailTxtBox\" class=\"textAllAnimals\">");
	printf("<h2 class=\"animaltitle\">&nbsp&nbsp%s </h2>", line_title);
	printf(" <ul class = \"a\">");
	printf(" <li><b>Overview:</b><br> %s </li><br><br>", line_overview);
	printf(" <li><b>Characteristics : </b><br> %s </li><br><br>", line_characteristics);
	printf(" <li><b>Special Traits : </b><br> %s </li></div>", line_special_traits);
	printf("</div></td></tr></div>");
	

}

File: zoo_project_cgi_php/animals.php
<!--
FILE			      : animals.php
PROJECT			    : A-03 : Animal-Serve – website developed in CGI and PHP (PROG2001)
FIRST VERSION	  : 2020-10-24 (Rev.04)
AUTHORS			    : Dusan Sasic & Kevin Downer
DESCRIPTION		  : The PHP code for the Animal Zoo project.
-->
<!DOCTYPE html>

  <?php
    // prep the elements required 
    // User name
    $name = $_POST['in_name'];	
    // Animal selection
    $animal = $_POST['combo_animals'];

    //Animal title 
    $title;
    // Overview animal text
    $line_overview;
    // Characteristics animal text
    $line_characteristics;
    // Traits animal text
    $line_special_traits;


    // external path to the image folder and what extension type to search
    $path_img = "theZoo/" . $animal . ".jpg";
    // external path to the text folder and what extension type to serach
    $path_text= "theZoo/" . $animal . ".txt";


    // Variable that determens which string will store which line from .txt file
    $i = 0;
    

    // open and read the external text files
    if ($file = fopen($path_text, "r")) 
    {
        // While the end of file is not reached 
        while(!feof($file)) 
        {
        // process line by line from the text file 
        $textperline = fgets($file);
            if($i == 0)
                $title = $textperline;
            else if($i == 1)
                // Save the first line in line_overview variable
                $line_overview = $textperline;
            if($i == 2)
                // Save the second line in line_characteristics variable
                $line_characteristics = $textperline;
            if($i == 3)
                // Save the third line in line_special_traits variable
                $line_special_traits = $textperline;
            $i++;
        }
    // finished with text file, safely close it
    fclose($file);
    }
  ?>


  <html>

  <head>
    <title>Animals</title>
    <link rel="stylesheet" href="ext-css-style-zoo.css">
  </head>

  <body>
   <!--  NOTE:  This code was changed to accomodate the html requirements, needs to be merged into CGI --> 
  <div class="helloName" align="center"><h1> Hello <?php print($name)?>, Look If You Dare!</h1></div>
  <div>
  <table align="center">
  <tr>
  <td  align="right" width="50%"> 
  <div class="pictureAnimal"width: 50%; margin: 0 auto;><img src="<?php echo($path_img)?>"></img></div>
  </td>
  <td  align="left" width="50%">
  <div id="detailTxtBox" class="textAllAnimals">
  
    <div class="content">
    <h2 class="animaltitle"><?php echo $title ?></h2> 
    <ul class="a">
      <li><b>Overview:</b><br><?php echo $line_overview ?></li>
      <br><br>
      <li><b>Characteristics:</b><br><?php echo $line_characteristics ?></li>
      <br><br>
      <li><b>Special Traits:</b><br><?php echo $line_special_traits ?></li>
    </div>

  </div>
  </td>
  </tr>
  </div>

  

  </body>

</html>